File: 0001_0100/0020/0020.py
def getLine(f):
    t = f.read(8192)
    k = f.read(1)
    while k != "" and k in "0123456789":
        t += k
        k = f.read(1)
    return t

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    f.readline()   
    m = c = 1
    lres = 0
    t = getLine(f)
    s = (int(x) for x in t.split())
    last = next(s)
    while t:
        for cur in s:
            res = 1 if cur > last else -1 if cur < last else 0
            if res == lres or res == 0:
                if c > m:
                    m = c
                c = 0 if res == 0 else 1
            c += 1
            lres, last = res, cur
        t = getLine(f)
        s = (int(x) for x in t.split())
    if c > m:
        m = c
    q.write(str(m))

File: 0001_0100/0020/0020.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	q, _ := os.Create("output.txt")
	content, _ := ioutil.ReadFile("input.txt")
	arr := strings.Fields(string(content))
	max, lres, c := 1, 0, 1
	n, _ := strconv.Atoi(arr[0])
	last, _ := strconv.Atoi(arr[1])
	for i, res := 2, 0; i <= n; i++ {
		cur, _ := strconv.Atoi(arr[i])
		if cur < last {
			res = -1
		} else if cur > last {
			res = 1
		}
		if res == lres || res == 0 {
			if c > max {
				max = c
			}
			if res == 0 {
				c = 0
			} else {
				c = 1
			}
		}
		c++
		lres = res
		last = cur
	}
	if c > max {
		max = c
	}
	fmt.Fprint(q, max)
}
